javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+
+3. Local Modules. 
+
+It is important to note that if the specified module name has a `./`, a `/` or a `../`, it will look for the directory/file in the given path. Please note that it will match the `.js`, `.json`, `.mjs`, `.cjs`, `.wasm` and `*.node` extensions.
+
+Here are some examples of popular Node built-in modules: `HTTP`, `URL`, `path` and `os`. For a more complete list, see this nicely laid out [W3School Reference](https://www.w3schools.com/nodejs/ref_modules.asp).
+
+Now, in the [guided practice below](#guided-practice), you will be able to test the `require` function.
+
+#### What is import?
+
+The `import` statement was first introduced in ES6, in 2015. It is a *static* import statement, which basically means that it will only bring in what has been exclusively and specifically exported from another module.
+
+The syntax is also quite different, as the module path must be a string literal, whereas when we use `require`, the module path can be dynamic.
+
+For example, with `require`, this will work:
+
+```javascript
+const name = "module2"; 
+const obj = require("./" + name);
+```
+
+However, due to the static aspect of `import`, you cannot do:
+
+```javascript
+// THIS WILL NOT WORK
+const name = "module2"; 
+import * from "./" + name;
+```
+
+You can also import multiple functions with the `import` statement. These are called **named imports**. To do so, you simply include the functions in curly braces.
+
+For example:
+
+```javascript
+// Named imports
+import {firstMessage, secondMessage} from './my-module.mjs';
+```
+
+You can also import a complete namespace. To do so, you simply use the following syntax:
+
+```javascript
+// Namespace import
+import * as myModule from './my-module.mjs';
+```
+
+If you simply import the name of the module, you will get the default export. This is common in libraries such as React:
+
+```javascript
+// Default import
+import myModule from './my-module.mjs';
+```
+
+And, of course, you can combine these:
+
+```javascript
+// Combinations:
+import myModule, {firstMessage, secondMessage} from './my-module.mjs';
+```
+
+Finally, you can also refer to an ES Module (if you need more information on modules, see the reference in the [supplement materials](supplement-materials)) This means that you can export objects and methods you have created.
+
+For example:
+
+```javascript
+module.exports = 'This is a big export!';
+```
+
+As with `require`, in the [guided practice](#guided-practice), you will be able to test all of this.
+
+#### Difference between import and require
+
+While there are many differences, which are detailed in the chart below, in my opinion, the two major difference between `require` and `import` is first that the `require` one will automatically scan node_modules to find modules whereas because it comes from ES6, `import` will not. Second, `import` won't support node_modules since it is based on ES6, but `require` will. Effectively, `import` must specify the path of the module.
```

Comment:
This paragraph is a little confusing so please rewrite and focus on the important difference for this curriculum is:

1. Use `import` and `export` when it's been configured for you, such as in a `create-react-app` app
1. Use `require` and `module.exports` everywhere else

Those other details aren't necessary.

Reply:
perfect!
done in [c89273b](https://github.com/Techtonica/curriculum/pull/1414/commits/c89273b844a10c1f67a1d0886a773f18a1c327e8)